### PR TITLE
Toolkit dropdown menu

### LINKED ISSUE #3985: Fix the dropdown menu for toolkit page
### Dependency
-  #3744 (its the issue that makes the component)
- #3256 

### Overview

During recent audits, the team realized that the drop-down menu design is not uniform. In an attempt to unify all drop-down menus the design team has redesigned the drop-down menu. This issue is to implement the new uniform design specifically on the toolkit page using the component designed for the project page.

### Action Items

- [x] Update the drop-down menu on the toolkit page to reflect the design in Figma using the component designed for the project page.
  - [x] You do not need to develop the search bar function.  The goal is to make this component reusable on multiple pages with different filters specified
- [x] Make the appropriate changes to the CSS, HTML, and JS as you need to
- [x] Ensure that changes to the component on the Toolkit page do not break pages in other pages (i.e. Projects Page, Wins, etc.)

### Resources/Instructions
- [final design](https://www.figma.com/file/0RRPy1Ph7HafI3qOITg0Mr/Hack-for-LA-Website?node-id=24519%3A284897) 
- for a reference a link to related design issue: #3626 
- in the elements folder https://github.com/hackforla/website/tree/gh-pages/_sass/elements you should find _dropdown_filters.SCSS that have been modified and should be ready to use
- in the [pages](https://github.com/hackforla/website/tree/gh-pages/pages) you should find the toolkit page https://github.com/hackforla/website/blob/gh-pages/pages/toolkit.html
- in the [assets/js](https://github.com/hackforla/website/tree/gh-pages/assets/js) you should find toolkit
- how to code for re-useability [Standardized Component](https://github.com/hackforla/website/blob/gh-pages/assets/js/toolkit.js) 
- write  up on [accessibility ](https://github.com/hackforla/website/issues/2173#issuecomment-1279768721)  (not specifically for drop down menus)

### PR DESCRIPTION
**Pull request by kwangric**
Fixes #3985

### What changes did you make and why did you make them ?

  - **Main Change:** Added a dropdown menu to toolkit page to allow users to filter guide cards based on different parameters. This includes adding the dropdown menu into the toolkit HTML `current-guides.html` and adding the filter functionality to the JS page `toolkit.js`. The filters should function correctly on desktop and mobile.
  - Refactored how the toolkit HTML was set up in order to be consistent with the Projects page (see `toolkit.html` & `current-guides.html`.
  - Changed `guide-card.html` to include all the properties from the guide-pages (status, practice areas, etc.) to allow the filter to function.
  - Changed the CSS in `_sass/components/_toolkit.scss` to match the Figma design and to be consistent with the current Projects page.
  - Even though the current guide cards in the repo do not contain Resource Type and Technologies properties as shown in the Figma filters, they will work with the new filters when added in the future (see [comments](https://github.com/hackforla/website/issues/3985#issuecomment-1555359253) in issue).
  - Verified that the changes do not affect or break other pages.

### Screenshots of Proposed Changes Of The Website  (if any, please do not screen shot code changes)

<details>
<summary>Visuals before changes are applied</summary>

### Desktop View (Before)
image

### Mobile View (Before)
image

</details>

<details>
<summary>Visuals after changes are applied</summary>
 
### Desktop View (After)
image

### Mobile View (After)
image

</details>

### CODE REVIEW

**Comment by kwangric** [comment]
While browsing the project board I noticed issue [#4753](https://github.com/hackforla/website/issues/4753), which noted that the placement of the red filter tags on the Projects page was unintentionally changed with a recent pull request and is currently being reverted. I had thought that the change was intentional when I was working on the issue, but I've moved the tags back to the original location with my latest commit.

<details>
<summary>New Filter Tag Location</summary>

image

</details>

**Comment by chrismenke45** [comment]
Review ETA: 6/6 EOD
Availability: 8am-1pm Monday

**Comment by chrismenke45** [comment]
@kwangric It looks really good. I think everything works as intended, but would it be possible for you to pull the most recent gh-pages into this branch. The issue in #4753 has been fixed and I just want to make sure that the fix won't affect your filters and your code won't affect those filters since they all use the same classes.

**Comment by kwangric** [comment]
@chrismenke45, branch has been merged with the latest version of gh-pages.

**Comment by Adastros** [comment]
Availability (PST): 10 - 11 PM 05JUN2023 - 07JUN2023
Review ETA (PST): End of day 07JUN2023

**Review by chrismenke45 — DISMISSED** [review]
@kwangric All the changes look good! The functionality is great and I don't see any effects on other pages.

**Review by Adastros — CHANGES_REQUESTED** [review]
Hi @kwangric, good job on completing the issue! 

- The filters work as expected, including the "Resource Type"  and " Technologies" filters
- The correct guide pages appear when selecting the filters
- The filter component behaves as expected when transitioning between mobile and desktop views
- The other components on the toolkit page are unaffected
- The other filters on the project, wins, and home pages do not seem to be affected by the change
- Works on both Firefox and Chrome

The only things I would request to be changed right now are as follows:

- [x] very minor but could you remove the issue # from the title of the pull request? This is to avoid any confusion on what # is associated with the pull request
- [x] I noticed that text in the filter tags can overflow the parent container. Can you update the styling for this to match how the filter on the projects page handles overflow for the filter tags? 

    <details>

    image

    </details>

File: assets/js/toolkit.js [hunk 1]
```diff
@@ -1,47 +1,554 @@
...
+            ${noResultsMessageComponent(filterParams)}
+    </div>
+    <div>
+        <h3>Search Tips</h3>
+        <ul>
+            <li><p>Broaden your search by removing some of the filters</p></li>
+        </ul>
+    </div>
+    `
+}
```
Comment (Adastros) [inline assets/js/toolkit.js:518]: I noticed that this feature of the filter component was added. Looking at `current-projects.js` and the [guide ](https://github.com/hackforla/website/issues/3744)@chrismenke45 drafted, it doesn't seem like it wasn't originally part of the filter component code. I think it's a great and much needed addition but it does make me me wonder if we can include it within the scope of the issue. 

Looking at the filter on the projects page, the messages don't appear when none of the projects fit the selected filters.

Thoughts @chrismenke45?

<details>
<summary>Toolkit Page</summary>

image

</details>

<details>
<summary>Projects Page</summary>

image

</details>

**Comment by Adastros** [comment]
I also had thoughts regarding the code in `toolkit.js`. Looking through it and reading this [guide](https://github.com/hackforla/website/issues/3744), the code deviates from `current-projects.js` a bit (differently named functions, different methods of sorting arrays, etc.). There's nothing wrong with that and I don't want to undo all the work @kwangric did; I'm not sure how strictly we want `toolkit.js` to align with the code in `current-projects.js` and (possibly) the other filters on the site. This could complicate things down the line if we have to change something on the filters since we'll need to take a look at differently coded filters with similar functionality. 

Any thoughts @chrismenke45 and @kwangric?

**Comment by Adastros** [comment]
I noticed a few other things with the filters in general while reviewing this pull request but it doesn't affect the approval status of this pull request. Below are the findings. Issues to fix them should be created if they haven't been created yet.

1. Depending on how many items are in a filter category, when the "view all" text is clicked on the scroll bar will appear but will not be scroll-able. It isn't appealing visually and may affect the UI experience if the user sees the scrollbar but isn't able to scroll it or thinks that there are missing/ unreachable filter items. Can be seen on projects filter too.

    <details>
    <summary>Firefox Screenshot</summary>
    It's a difficult to see on Firefox due to the background color

    image

    </details>

    <details>
    <summary>Chrome Screenshot</summary>

    image

    </details>

2. This was discussed in last Sunday's meeting for the projects page filter. The text for some filter tags overflows the container. The issue that will be created to fix it for the projects page should also take the filters in in the toolkit into account. Note that one of the contributors has a long name. A solution will need to be determined on how to handle this case.

    <details>

    image

    </details>

3. The scrolling behavior of the mobile-view of the filter component does not function properly. I can scroll the toolkit page while the filter is active. This will affect the UI experience since users may unintentionally end up on a different part of the page while selecting the filters. Can be seen on projects filter too. 
4. The bottom portion of the filters on mobile is cutoff/covered by the div containing the cancel and apply buttons. Can be seen on projects filter too.
5. When toggling the drop down arrows for the filter categories without moving the cursor, the arrows are highlighted/selected by the cursor. This can hamper the UI experience. 

    <details>

    image

    </details>

6. Adding onto # 5, the selectable/interactable area for the arrow changes when it is pointed down or up. The arrow should be  be updated so that the selectable area is consistent in the up and down positions. Can be seen on projects filter too.

    <details>

    image

    </details>

**Comment by kwangric** [comment]
@Adastros I should have the changes you requested done by tomorrow. 

As for the difference between `current-project.js`, since the projects and guide pages are set up differently in the repository, I had to pull the information from the files in a different way. From there, I tried to reuse some of the same code from `current-projects.js` but set it up in a way that was intuitive to me.

File: assets/js/toolkit.js (hunk 1 above)
Comment (Adastros) [reply assets/js/toolkit.js:518]: Looks like an ER was created to address this for the projects filter. See #4769. This feature might need to be removed for this pull request and addressed in the issue that will be created for ER #4769. 

Please hold on doing anything for this portion of the code.

**Comment by kwangric** [comment]
@Adastros I made the changes that you requested. 

One thing I noticed about the project page is that the project filter tags also have the same overflow issue, but it's just hard to see since the background is also white.

**Comment by Adastros** [comment]
> @Adastros I made the changes that you requested.
> 
> One thing I noticed about the project page is that the project filter tags also have the same overflow issue, but it's just hard to see since the background is also white.

Thanks for making the changes! Good job on catching that! I'm okay with the changes since it only affects the "Professional Development" and contributor Ahdithan Uthayakumar Tarang Gupta tags. I think they can be shortened in a separate issue if we want to make them fit. The overflow issue on the projects page will be addressed in issue #4784. 

Before I approve this pull request, could you comment in ER issue #4769 on how the "no results" feature works? You can write about the relevant functions you added, how they work together, and what changes should/can be made to make the code reusable for the toolkit and projects page filters. It would be good to reference this [guide](https://github.com/hackforla/website/issues/3744) from the projects page filter to explain how a dev would try to refactor your code to make it work on the projects filter page too.

<details>
<summary>Toolkit Page Filter - No Results</summary>

image

</details>

File: assets/js/toolkit.js (hunk 1 above)
Comment (Adastros) [reply assets/js/toolkit.js:518]: See [comment](https://github.com/hackforla/website/issues/4769#issuecomment-1588758037) and [this comment](https://github.com/hackforla/website/pull/4764#issuecomment-1590553134) for resolution.

**Comment by kwangric** [comment]
@Adastros I just left a comment on the ER issue with how the feature works and some steps on how it could be implemented on another page. Please let me know if anything else is needed.

**Review by Adastros — APPROVED** [review]
Thanks for being patient and accommodating @kwangric. The note on ER issue #4769 describes covers the "no result" feature well.

**Comment by chrismenke45** [comment]
My apologies. I hadn't realized I'd be rerequested to review this while I was out of town.
ETA: 6/18 EOD

**Review by chrismenke45 — APPROVED** [review]
Looks good @kwangric ! @Adastros you brought up a lot of good points in your review. Its good to see any lingering problems with the filters will be addressed with those ERs. Sorry for my lack of responses during the review process. I was out of town and when I got back and didn't notice.

**Pull request merged**